src/puppetlabs/puppetdb/cli/services.clj

```diff
@@ -259,83 +259,81 @@
+                            mq/activemq-connection-factory)
+            mq-connection (doto (.createConnection mq-factory)
+                            (.setExceptionListener
+                             (reify ExceptionListener
+                               (onException [this ex]
+                                 (log/error ex "service queue connection error"))))
+                            .start)
+            globals {:scf-read-db read-db
+                     :scf-write-db write-db
+                     :authorizer (some-> certificate-whitelist build-whitelist-authorizer)
+                     :update-server update-server
+                     :product-name product-name
+                     :url-prefix url-prefix
+                     :discard-dir (.getAbsolutePath discard-dir)
+                     :mq-addr mq-addr
+                     :mq-dest mq-endpoint
+                     :mq-threads threads
+                     :catalog-hash-debug-dir catalog-hash-debug-dir
+                     :command-mq {:connection mq-connection
+                                  :endpoint mq-endpoint}}
+            updater (when-not disable-update-checking
+                      (future (shutdown-on-error
+                               (fn [] (maybe-check-for-updates puppet-enterprise? update-server read-db))
+                               shutdown-updater)))]
+        (do (log/info "Starting query server")
+            (add-ring-handler (server/build-app globals)))
+
+        ;; Pretty much this helper just knows our job-pool and gc-interval
+        (let [job-pool (mk-pool)
+              gc-interval-millis (to-millis gc-interval)
+              gc-task #(interspaced gc-interval-millis % job-pool)
+              db-maintenance-tasks (fn []
+                                     (let [seconds-pos? (comp pos? to-seconds)] 
+                                       (when (seconds-pos? node-ttl) (auto-expire-nodes! node-ttl write-db mq-connection))
+                                       (when (seconds-pos? node-purge-ttl) (purge-nodes! node-purge-ttl write-db)) 
+                                       (when (seconds-pos? report-ttl) (sweep-reports! report-ttl write-db))
+                                       ;; Order is important here to ensure
+                                       ;; anything referencing an env or resource
+                                       ;; param is purged first
+                                       (garbage-collect! write-db)))]
+          ;; Run database maintenance tasks seqentially to avoid
+          ;; competition. Each task must handle its own errors.
+          (gc-task db-maintenance-tasks)
+          (gc-task (fn [] (compress-dlo! dlo-compression-threshold discard-dir))))
+        (assoc context
+               :broker broker
+               :mq-factory mq-factory
+               :mq-connection mq-connection
+               :shared-globals globals
+               :updater updater)))))
```

Comment:
I'm having a hard time understanding what has changed in this function. Is it just a refactor?